app/platform/metrics.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package platform
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"runtime"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/v6/einterfaces"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/pkg/errors"
+)
+
+const TimeToWaitForConnectionsToCloseOnServerShutdown = time.Second
+
+type platformMetrics struct {
+	server *http.Server
+	router *mux.Router
+	lock   sync.Mutex
+
+	metricsImpl einterfaces.MetricsInterface
+
+	cfgFn func() *model.Config
+}
+
+// resetMetrics resets the metrics server. Clears the metrics if the metrics are disabled by the config.
+func (ps *PlatformService) resetMetrics(metricsImpl einterfaces.MetricsInterface, cfgFn func() *model.Config) {
+	if !*cfgFn().MetricsSettings.Enable {
+		ps.metrics = nil
+		return
+	}
+
+	ps.metrics = &platformMetrics{
+		cfgFn:       cfgFn,
+		metricsImpl: metricsImpl,
+	}
+
+	ps.metrics.stopMetricsServer()
+
+	if err := ps.metrics.initMetricsRouter(); err != nil {
+		mlog.Error("Error initiating metrics router.", mlog.Err(err))
```

Comment:
Let's avoid using global logger. We can pass the logger instance from the server and add it to a field in the struct and use that.

Even better to actually return the error since `platform.New` actually returns an error